find/main/routes.py

```diff
@@ -111,10 +112,10 @@ def download():
             try:
                 trigger_async_download(query_params)
                 current_app.logger.info(
-                    "Request for download by {user_id} with {query_params}",
+                    "Request for download by user with ID: {user_id} with query params: {query_params}",
                     extra={
                         "user_id": g.account_id,
-                        "query_params": query_params_without_email_address,
+                        "query_params": json.dumps(query_params_without_email_address),
```

Comment:
I think there's a bit of miscommunication on my end here.

The ticket said something like "don't log mutable objects like lists and dicts". The reasoning behind this is because it's easier for those objects to, at some point in the future, have extra data inserted into them that might contain personal data we don't want to log. It's not because lists and dicts are inherently bad objects that can't be logged but strings and ints are OK.

So converting this dict to a string doesn't really meet our objectives - we'll still be logging whatever is inside that dict.

I think the actual principle here is more - don't like things where it isn't immediately obvious what is being logged, and where changes elsewhere might easily change what is being logged here.

This log call is _basically_ fine already because we explicitly craft the logged data in very close proximity to the logging call, so it would be hard for extra things to sneak in. However an even safer approach might be an allowlist rather than a blocklist:

```python
query_params_without_email_address = {k: v for k, v in query_params.items() if k in ['organisations', 'regions', 'funds', 'outcome_categories', 'rp_start', 'rp_end']}
try:
    trigger_async_download(query_params)
    current_app.logger.info(
        "Request for download by user with ID: {user_id} with query params: {query_params}",
        extra={
            "user_id": g.account_id,
            "query_params": json.dumps(query_params_without_email_address),
            "request_type": "download",
        },
    )
```

Reply:
Makes sense, thank you! Changed it.